vignettes/miaViz.Rmd

```diff
@@ -350,101 +350,73 @@ tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray",
+p + geom_path(aes(x=X, y=Y, group=subject), 
               arrow=arrow(length = unit(0.1, "inches")),
-              data = subset(df, subject %in% selected.subjects))+ 
-    labs(title = "All trajectories with two time points")
-
-print(p1)
-```
-
-Filtering by trajectory length and displaying top 10%:
-
-```{r}
-# calculating trajectory lengths as sum of euclidean distances (based on the 
-# ordination coordinates) of each successive time points in each series.
-df <- merge(df,
-               df %>% group_by(subject) %>% group_modify(~{
-                   # distance matrix between points based on PC1 and PC2
-                   mat <- as.matrix(dist(.x[,c("PC1", "PC2")]));
-                   # summing up only the distance between successive points
-                   data.frame(change=sum(mat[row(mat) == col(mat) + 1]))
-               }),
-            by.y="subject") %>% arrange(time)
-# plot
-p1 <- plotReducedDim(tse, dimred = "PCoA_BC") +
-         labs(title="PCoA - Bray-Curtis")+
-    theme(plot.title = element_text(hjust = 0.5)) +
-    geom_path(aes(x=PC1, y=PC2, color=change, group=subject),
-              data = subset(df, subject %in% selected.subjects) %>%
-                  top_frac(0.1, change),
-              arrow=arrow(length = unit(0.1, "inches")))+
-    labs(title = "Top 10% trajectories having highest change from time point one to two")+
-    scale_color_gradient2(low="white", high="red")
-print(p1)
+              data = subset(p$data, subject %in% selected.subjects))+
+    labs(title = "All trajectories with two time points")+
+    theme(plot.title = element_text(hjust = 0.5))
 ```
 
-Filtering by trajectory length and displaying top 10% along with nationality:
+Filtering the two time point trajectories by divergence and displaying top 10%:
 
 ```{r}
+library(miaTime)
+# calculating step wise divergence based on the microbial profiles
+tse <- getStepwiseDivergence(tse, group = "subject", time_field = "time")
+# adding metadata for the plot
+p$data <- cbind(p$data, colData(tse)[,"time_divergence", drop=FALSE])
+# retrieving the top 10% divergent subjects having two time points
+top.selected.subjects <- p$data %>% filter(subject %in% selected.subjects) %>%
+    top_frac(0.1, time_divergence) %>% select(subject) %>% .[[1]]
 # plot
-p1 <- plotReducedDim(tse, dimred = "PCoA_BC") +
-         labs(title="PCoA - Bray-Curtis")+
-    theme(plot.title = element_text(hjust = 0.5)) +
-    geom_path(aes(x=PC1, y=PC2, color=nationality, group=subject),
-              data = subset(df, subject %in% selected.subjects) ,
-              arrow=arrow(length = unit(0.1, "inches")))+
-    labs(title = "Trajectories with two time points and nationality")
-print(p1)
+p + geom_path(aes(x=X, y=Y,
+                  color=time_divergence, group=subject),
+              data = subset(p$data, subject %in% top.selected.subjects) %>% 
+                  arrange(desc(time)), 
+              arrow=arrow(length = unit(0.1, "inches"), ends = "first"))+
```

Comment:
I think this is fine. Just add a comment "time points are in descending order, --> specify arrows to point the second time point" or something like that

Reply:
As far as I understand, the divergence is not available for start or end of the segment; divergence is something that is calculated between two time points, not at first or second time point. How divergence between time points 1 and 2 be different at the start (1) and end (2) time points?

Reply:
Yes I agree. However storing results back into colData the divergence between two consecutive time points will be stored at the second point (the following is an example):

![image](https://user-images.githubusercontent.com/86964562/209793390-9b240539-6aeb-4797-96da-54698ddce908.png)

Reply:
Right. This is true and logical since divergence marks something that only exists after the second observation.

Ok I also do not know if there is any better solution than just draw the arrows backwards. 

Just make sure to explain it clearly in the comments that this is how geom_path works and hence we do it this way.

Reply:
Later we might consider a wrapper that makes stuff automatically but we do not know yet how essential that would be to have, so let us leave it to the example for now.